lib/matplotlib/colorbar.py

```diff
@@ -1465,7 +1482,9 @@ def make_axes_gridspec(parent, *, location=None, orientation=None,
 
     fig = parent.get_figure()
     cax = fig.add_subplot(ss_cb, label="<colorbar>")
-    cax.set_aspect(aspect, anchor=loc_settings["anchor"], adjustable='box')
+    cax.set_anchor(anchor)
```

Comment:
Was `loc_settings["anchor"]` an error?

Reply:
Yeah, I assume that `anchor` was just ignored if passed as a kwarg.  I'm not sure what happens if it _is_ passed as a kwarg.

Reply:
Strange, it should just go through to `set_anchor` if it's not `None`.

Reply:
I think what is happening here is correct, and the old code was ignoring the user's specification of `anchor`, but maybe I'm not following...